### PR TITLE
Adds `script_env` support

### LINKED ISSUE #64: `/build/script` needs to be upgraded
The `/build/script` section is currently not being upgraded. In simple scenarios, this works out fine. In complex situations, we need to use the new [Script object](https://github.com/conda/ceps/blob/main/cep-14.md#script-section)

This will be a pretty involved upgrade process that will need to be heavily conditionalized. From a brief search, there are already many variants AND we have to parse-out shell variable syntax in the existing [script_env](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#use-environment-variables) field.

### PR DESCRIPTION
**Pull request by schuylermartin45**
- Adds basic support for upgrading a recipe that contains a `script_env` section.
  - NOTE: There is one edge case that is not currently supported and is likely best left for a human to fix. The conversion process will flag this with an error.
- Adds unit tests

Fixes #64

### CODE REVIEW

File: conda_recipe_manager/parser/_utils.py [hunk 1]
```diff
@@ -188,3 +188,15 @@ def dedupe_and_preserve_order(l: list[H]) -> list[H]:
 
     """
     return list(cast(dict[H, None], dict.fromkeys(l)))
+
+
+def set_key_conditionally(dictionary: dict[str, JsonType], key: str, value: JsonType) -> None:
+    """
+    Convenience function that conditionally includes a key-value pair in a dictionary if the value is truthy.
+    Great for cheating McCabe ratings in complex JSON/YAML operations!
```
Comment (beeankha) [inline conda_recipe_manager/parser/_utils.py:196]: 😆

**Review by beeankha — APPROVED** [review]
